client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -88,20 +102,61 @@ class MappedDomainType extends React.Component {
 		return (
 			<React.Fragment>
 				<div>
-					<p>{ primaryMessage }</p>
-					{ ! isSubdomain( domain.name ) && (
-						<ul className="mapped-domain-type__name-server-list">
-							{ WPCOM_DEFAULT_NAMESERVERS.map( ( nameServer ) => {
-								return <li key={ nameServer }>{ nameServer }</li>;
-							} ) }
-						</ul>
-					) }
+					<p>{ setupInstructionsMessage }</p>
+					{ this.renderRecommendedSetupMessage( primaryMessage ) }
+					{ domain.aRecordsRequiredForMapping && this.renderARecordsMappingMessage() }
 				</div>
 				<div className="mapped-domain-type__small-message">{ secondaryMessage }</div>
 			</React.Fragment>
 		);
 	}
 
+	renderRecommendedSetupMessage( primaryMessage ) {
+		const { domain, translate } = this.props;
+
+		return (
+			<FoldableFAQ
+				id="recommended-mapping-setup"
+				question={ translate( 'Recommended setup' ) }
+				expanded
+			>
+				<p>{ primaryMessage }</p>
+				{ ! isSubdomain( domain.name ) && (
+					<ul className="mapped-domain-type__name-server-list">
+						{ WPCOM_DEFAULT_NAMESERVERS.map( ( nameServer ) => {
+							return <li key={ nameServer }>{ nameServer }</li>;
+						} ) }
+					</ul>
+				) }
+			</FoldableFAQ>
+		);
+	}
+
+	renderARecordsMappingMessage() {
+		const { domain, translate } = this.props;
+
+		const generateLinkTo = ( href ) => (
+			<a href={ href } target="_blank" rel="noopener noreferrer" />
+		);
```

Comment:
We could probably make that function a method of that component to avoid duplicating it. As an aside, we usually prefix methods that output markup with `render`.